x-pack/plugins/observability/server/services/slo/summary_transform_generator/generators/common.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ALL_VALUE } from '@kbn/slo-schema';
+import { SLO } from '../../../../domain/models/slo';
+
+export const getGroupBy = (slo: SLO) => {
+  const groupings =
+    slo.groupBy !== '' && slo.groupBy !== ALL_VALUE
+      ? [slo.groupBy].flat().reduce((acc, field) => {
+          return {
+            ...acc,
+            [`slo.groupings.${field}`]: {
+              terms: {
+                field: `slo.groupings.${field}`,
+              },
+            },
+          };
+        }, {})
+      : {};
+
+  return {
+    'slo.id': {
+      terms: {
+        field: 'slo.id',
+      },
+    },
+    'slo.revision': {
+      terms: {
+        field: 'slo.revision',
+      },
+    },
+    'slo.instanceId': {
```

Comment:
So now with the multiple groupings, how is instance id set? Is it still only based on one value?

Reply:
We haven't changed the implementation at the moment, e.g. we don't handle mutliple groupBy. Only the mapping has been changed to handle multiple group by in the future. Then we will still have to change the way we compute instanceId (fingerprint or concat values..).

Reply:
I tried to do it in this PR, but using fingerprint was pulling too many changes in APIs and frontend that I did not want to do at this time